tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/bulkhead/BulkheadFutureTest.java

```diff
@@ -77,126 +83,121 @@ public void beforeTest(final ITestContext testContext) {
     }
-  
+
     /**
      * Tests that the Future that is returned from a asynchronous bulkhead can
      * be queried for Done OK after a goodpath get with timeout and also
      * multiple gets can be called ok. This test is for the annotation at a
      * Class level.
      */
-    @Test()
+    @Test
     public void testBulkheadClassAsynchFutureDoneAfterGet() {
-
         Checker fc = new FutureChecker(SHORT_TIME);
         Future<String> result = null;
 
         try {
             result = bhBeanClassAsynchronousDefault.test(fc);
         }
         catch (InterruptedException e1) {
-            Assert.fail("Unexpected interruption", e1);
+            fail("Unexpected interruption", e1);
         }
 
-        Assert.assertFalse(result.isDone(), "Future reporting Done when not");
+        assertFalse(result.isDone(), "Future reporting Done when not");
         try {
             String r;
-            Assert.assertTrue("GET_TO.".equals(r = result.get(1, TimeUnit.SECONDS)), r);
-            Assert.assertTrue("GET_TO.GET.".equals(r = result.get()), r);
+            assertEquals(r = result.get(1, TimeUnit.SECONDS), "GET_TO.", r);
+            assertEquals(r = result.get(), "GET_TO.GET.", r);
 
         }
         catch (Throwable t) {
-            Assert.assertNull(t);
+            assertNull(t);
         }
-        Assert.assertTrue(result.isDone(), "Future done not reporting true");
+        assertTrue(result.isDone(), "Future done not reporting true");
     }
 
     /**
      * Tests that the Future that is returned from a asynchronous bulkhead can
      * be queried for Done OK when get() is not called. This test is for the
      * annotation at a Class level.
      */
-    @Test()
+    @Test
     public void testBulkheadClassAsynchFutureDoneWithoutGet() {
-
         Checker fc = new FutureChecker(SHORT_TIME);
         Future<String> result = null;
+
         try {
             result = bhBeanMethodAsynchronousDefault.test(fc);
         }
         catch (InterruptedException e1) {
-            Assert.fail("Unexpected interruption", e1);
+            fail("Unexpected interruption", e1);
         }
-        Assert.assertFalse(result.isDone(), "Future reporting Done when not");
+
+        assertFalse(result.isDone(), "Future reporting Done when not");
+        waitingSimulation(result);
+        assertTrue(result.isDone(), "Future done not reporting true");
+    }
+
+    /**
+     * method that simulates the waiting time using awaitility library
+     * @param result future to simulate the waiting time
+     */
+    private void waitingSimulation(Future<String> result) {
         try {
-            Thread.sleep(SHORT_TIME + SHORT_TIME);
-            // Usually, we will not enter the loop, 
+            final int waitingInitialCondition = 0;
+
+            await().atMost(SHORT_TIME * 2, MILLISECONDS)
+                .untilAsserted(() -> assertEquals(waitingInitialCondition, 0));
```

Comment:
Actually, why do we even do this? We set a variable to 0 and then we wait up to 200 millis, until the variable is 0?

Reply:
So, we replaced this

```java
        Assert.assertFalse(result.isDone(), "Future reporting Done when not");
        try {
            Thread.sleep(SHORT_TIME + SHORT_TIME);
            // Usually, we will not enter the loop, 
            // but we are prepared to wait up to 100 SHORT_TIMES (10sec)
            int loops = 0;
            while(!result.isDone() && loops++ < 100) {
                Thread.sleep(SHORT_TIME);
            }
        }
        catch (Throwable t) {
            Assert.assertNull(t);
        }
        Assert.assertTrue(result.isDone(), "Future done not reporting true");
```

with this

```java
        assertFalse(result.isDone(), "Future reporting Done when not");
        waitingSimulation(result);
        assertTrue(result.isDone(), "Future done not reporting true");
```

I think the replacement should be

```
        assertFalse(result.isDone());
        await().atMost(100 * SHORT_TIME, MILLISECONDS).untilAsserted(() -> assertTrue(result.isDone());
```

It's short and conveys intent well, and doesn't need a `waitingSimulation` method (which still contains nonsensical code like `catch (Throwable t) { assertNull(t); }`).

Reply:
yes for me make sense and with that, we can remove the waitingSimulation method and make this more readable, I'll make the change and run the tests again thank you @Ladicek

Reply:
done